### ISSUE TITLE
Redundant Service Worker Bug (again?)

### ISSUE THREAD

**Issue by interglobalmedia** [gatsbyjs/gatsby#14813]
## Description

Describe the issue that you're seeing. I had a whole bunch of related issues inititially, but then did some plugin updates, removed .cache many times as well as public, as well as updating my Mac to 10.14.5, and now with any of my unrelated projects OR an gatsby starter themes I may download (and then open in Chrome), I get the following in the browser:

```
Unhandled Rejection (SyntaxError): Unexpected token < in JSON at position 0
XMLHttpRequest.request.onreadystatechange
/Users/mariacam/Development/interglobalmedia/webpack/bootstrap:52
  49 | } else {
  50 | 	// success
  51 | 	try {
> 52 | 		var update = JSON.parse(request.responseText);
  53 | 	} catch (e) {
  54 | 		reject(e);
  55 | 		return;
```

### Steps to reproduce

Clear steps describing how to reproduce the issue. Please please please link to a demo project if possible, this makes your issue _much_ easier to diagnose (seriously).

I use npx gatsby develop to get the server running. I do not have gatsby-cli installed globally nor gatsby installed globally. Only locally.
I have already pushed two sites to remote on Netlify where they work well with Chrome. Only within the last couple of days does nothing work LOCALLY. Link to one repo which has been hosted on Netlify (and made changes just a few days ago) is the following: https://github.com/interglobalmedia/interglobalmedia

### Expected result
The site to render properly in the browser.
What should happen?
The site should render properly in the browser.
### Actual result
```
Unhandled Rejection (SyntaxError): Unexpected token < in JSON at position 0
XMLHttpRequest.request.onreadystatechange
/Users/mariacam/Development/interglobalmedia/webpack/bootstrap:52
  49 | } else {
  50 | 	// success
  51 | 	try {
> 52 | 		var update = JSON.parse(request.responseText);
  53 | 	} catch (e) {
  54 | 		reject(e);
  55 | 		return;
```

And in the terminal console (npx gatsby develop):

```
You can now view interglobalmedia in the browser.

  http://localhost:8000/

View GraphiQL, an in-browser IDE, to explore your site's data and schema

  http://localhost:8000/___graphql

Note that the development build is not optimized.
To create a production build, use npm run build

info ℹ ｢wdm｣:
info ℹ ｢wdm｣: Compiled successfully.
Page not found /dev-404-page
```
What happened.
The site did not render.
### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

```
npx gatsby info --clipboard                                                                      ⏎ ✭

  System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.14.0 - ~/.nvm/versions/node/v11.14.0/bin/node
    npm: 6.7.0 - ~/.nvm/versions/node/v11.14.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  npmPackages:
    gatsby: ^2.4.6 => 2.4.6 
    gatsby-image: ^2.0.40 => 2.0.40 
    gatsby-plugin-emotion: ^4.0.6 => 4.0.6 
    gatsby-plugin-feed: ^2.1.1 => 2.1.1 
    gatsby-plugin-google-analytics: ^2.0.18 => 2.0.18 
    gatsby-plugin-gtag: ^1.0.10 => 1.0.10 
    gatsby-plugin-manifest: ^2.0.29 => 2.0.29 
    gatsby-plugin-offline: ^2.0.25 => 2.0.25 
    gatsby-plugin-react-helmet: ^3.0.12 => 3.0.12 
    gatsby-plugin-sass: ^2.0.11 => 2.0.11 
    gatsby-plugin-sharp: ^2.0.35 => 2.0.35 
    gatsby-plugin-sitemap: ^2.0.12 => 2.0.12 
    gatsby-plugin-twitter: ^2.0.13 => 2.0.13 
    gatsby-plugin-typography: ^2.2.13 => 2.2.13 
    gatsby-remark-copy-linked-files: ^2.0.12 => 2.0.12 
    gatsby-remark-images: ^2.0.6 => 2.0.6 
    gatsby-remark-prismjs: ^3.2.8 => 3.2.8 
    gatsby-remark-responsive-iframe: ^2.1.1 => 2.1.1 
    gatsby-remark-smartypants: ^2.0.9 => 2.0.9 
    gatsby-source-filesystem: ^2.0.32 => 2.0.32 
    gatsby-transformer-remark: ^2.3.11 => 2.3.11 
    gatsby-transformer-sharp: ^2.1.18 => 2.1.18 

```

**Comment by interglobalmedia** [comment]
I am  also getting the following locally for the referenced interglobalmedia project (repo) in the console:

`app.js:66 Warning: found one or more service workers present. If your site isn't behaving as expected, you might want to remove these. [ServiceWorkerRegistration]`

And a whole bunch of workbox messages. When I go into the Network tab, everything is coming up ok 200 status. And nothing is being blocked. Could this be a gatsby-plugin-offline issue? I never had any problems with the Manifest, workbox, etc. when I set up my own React workflows and used Chrome for development. And previously, my PWA set up passed in Lightbox with flying colors. Just days ago. This is really a mystery to me. The situation just seems so unstable.

**Comment by interglobalmedia** [comment]
And lastly, what is very new locally:

```
DevTools failed to parse SourceMap: http://localhost:8000/workbox-v3.6.3/workbox-sw.js.map
21:13:54.023 DevTools failed to parse SourceMap: http://localhost:8000/workbox-v3.6.3/workbox-core.dev.js.map
21:13:54.027 DevTools failed to parse SourceMap: http://localhost:8000/workbox-v3.6.3/workbox-precaching.dev.js.map
21:13:54.029 DevTools failed to parse SourceMap: http://localhost:8000/workbox-v3.6.3/workbox-routing.dev.js.map
21:13:54.029 DevTools failed to parse SourceMap: http://localhost:8000/workbox-v3.6.3/workbox-strategies.dev.js.map
```

And remotely at interglobalmedianetwork.com:

`The resource https://www.interglobalmedianetwork.com/static/d/604/path---offline-plugin-app-shell-fallback-a-30-c5a-BawJvyh36KKFwbrWPg4a4aYuc8.json was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.`

**Comment by interglobalmedia** [comment]
I solved the issue. I had been using @emotion/core @emotion/styled and was on version 10+. However, there is a major breaking change in version 10, and not a good idea to use anymore. So I changed over to styled components instead, AND in one of my deployed sites, interglobalmedia is the repo, I updated to the latest of gatsby everything, installed styled components and the necessary dependencies which I got from the Gatsby docs, removed emotion plugins, and added styled components to the gatsby-config.js file. Then I re-built successfully, ran npx gatsby develop, first opened in Firefox, and then in Chrome. Worked in both. Hope this might help others with this issue.
